File: Store.Client/src/components/product/product.ts
import { Component, Input } from '@angular/core';

interface Product {
  idxCode: string;
  idxCodeAlt: string;
  name: string | null;
  date: string | null;
  quantity: number | null;
  price: number | null;
}

@Component({
  selector: 'app-product',
  templateUrl: 'product.html',
  styleUrls: ['product.css']
})
export class ProductComponent {
  @Input() product: Product | null = null;

  getImagePath(): string | null {
    const coreFileName = "image_";
    var randomImageNumber = Math.floor((Math.random() * 9 + 1));
    return this.product ? `assets/images/${coreFileName}${randomImageNumber}.jpg` : null;
  }
}

File: Store.Server/Models/Product.cs
namespace Store.Server.Entities
{
    public class Product
    {
        public string? Name { get; set; }

        public string? IdxCode { get; set; }

        public string? IdxCodeAlt { get; set; }

        public DateTime? Date { get; set; } = default;

        public int Quantity { get; set; }

        public int Price { get; set; }
    }
}
